### ISSUE TITLE
PlatformCompatibilityAnalyzer returns spurious errors in AspNetCore sources

### ISSUE THREAD

**Issue by pranavkm** [dotnet/roslyn-analyzers#4209]
* Clone the AspNetCore repo
* Run `restore.cmd`
* cd src/SignalR
* startvs.cmd
* Open `WebSocketsTransport.cs`
* Remove the suppression: https://github.com/dotnet/aspnetcore/blob/master/src/SignalR/clients/csharp/Http.Connections.Client/src/Internal/WebSocketsTransport.cs#L145
* Notice that the analyzer complains about unsupported OS platform.

**Comment by buyaa-n** [comment]
Thank you @pranavkm for reporting the bug, i was able to narrow down with simple repro.
Invoking a functor breaking flow analysis: cc @mavasani 

```csharp
class Test
{
    Func<string> Greetings = () => "Hi!";
    
    void M1()
    {
        Greetings();
        if (OperatingSystem.IsBrowser())
        {
            SupportedOnBrowser();  // Failed, warns
        }
        else
        {
            UnsupportedOnBrowser(); // Failed, warns
        }
    }

    [SupportedOSPlatform("browser")]
    void SupportedOnBrowser() { }

    [UnsupportedOSPlatform("browser")]
    void UnsupportedOnBrowser() { }
}
```

**Comment by mavasani** [comment]
Fixed with https://github.com/dotnet/roslyn-analyzers/pull/4218#pullrequestreview-496015898
